Src/DependencyCollector/Shared/Implementation/ProfilerSqlProcessingBase.cs

```diff
@@ -253,6 +251,80 @@ private object OnBegin(object thisObj)
         }
 
         /// <summary>
+        ///  Common helper for all EndAsync Callbacks.
+        /// </summary>
+        /// <param name="taskObj">Returned task by the async method.</param>
+        /// <param name="thisObj">This object.</param>
+        private void OnEndAsync(object taskObj, object thisObj)
```

Comment:
no, that's our callbacks and we have an access to the return value i.e. Task only in OnEnd

---
In reply to: [97906094](https://github.com/Microsoft/ApplicationInsights-dotnet-server/pull/298#discussion_r97906094) [](ancestors = 97906094)